pkg/envoy/cds/cluster.go

```diff
@@ -25,6 +25,22 @@ const (
 	clusterConnectTimeout = 1 * time.Second
 )
 
+func getPassthroughCluster(upstreamSvc service.MeshService) (*xds_cluster.Cluster, error) {
+	HTTP2ProtocolOptions, err := envoy.GetHTTP2ProtocolOptions()
+	if err != nil {
+		return nil, err
+	}
+
+	return &xds_cluster.Cluster{
+		Name:                          upstreamSvc.String(),
+		ConnectTimeout:                ptypes.DurationProto(clusterConnectTimeout),
+		TypedExtensionProtocolOptions: HTTP2ProtocolOptions,
+		ClusterDiscoveryType:          &xds_cluster.Cluster_Type{Type: xds_cluster.Cluster_EDS},
+		EdsClusterConfig:              &xds_cluster.Cluster_EdsClusterConfig{EdsConfig: envoy.GetADSConfigSource()},
+		LbPolicy:                      xds_cluster.Cluster_ROUND_ROBIN,
+	}, nil
+}
```

Comment:
We have been referring to passthrough clusters as clusters that use original destination type cluster discovery (ex. with global egress).  Can you not use `getUpstreamServiceCluster()` for your use case?

Reply:
`getUpstreamServiceCluster`, sets the TLS context, which we don't want to do for SSL Passthrough.

Renamed to getSSLPasshthrough, but open to other suggestions.

In permissive mode, we also don't necessarily want to use the ORIGINAL_DST, since that will look for a cluster with a name where we don't have a DNS entry (ie: with the .global suffix)

Reply:
Could you instead reuse this function and pass a bool for `useTLS bool`?

Reply:
added variadic options, IMO the other way is more clean though.